src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -0,0 +1,490 @@
+        }
+
+        std::vector<std::vector<TopoDS_Wire>> wiresections;
+        for(TopoDS_Wire& wire : wires)
+            wiresections.emplace_back(1, wire);
+
+        //maybe we need a scaling law
+        Handle(Law_Function) scalinglaw;
+
+        //build all shells
+        std::vector<TopoDS_Shape> shells;
+        std::vector<TopoDS_Wire> frontwires, backwires;
+        for(std::vector<TopoDS_Wire>& wires : wiresections) {
+
+            BRepOffsetAPI_MakePipeShell mkPS(TopoDS::Wire(path));
+
+            mkPS.SetTolerance(Precision::Confusion());
+            mkPS.SetTransitionMode(BRepBuilderAPI_Transformed);
+
+            mkPS.SetMode(true);  //This is for frenet
+            //mkPipeShell.SetMode(TopoDS::Wire(auxpath), true);  // this is for two rails
+
+
+            if(!scalinglaw) {
+                for(TopoDS_Wire& wire : wires) {
+                    wire.Move(invObjLoc);
+                    mkPS.Add(wire);
+                }
+            }
+            else {
+                for(TopoDS_Wire& wire : wires)  {
+                    wire.Move(invObjLoc);
+                    mkPS.SetLaw(wire, scalinglaw);
+                }
+            }
+
+            if (!mkPS.IsReady())
+                return new App::DocumentObjectExecReturn("Pipe could not be built");
+
+            shells.push_back(mkPS.Shape());
+
+
+            if (!mkPS.Shape().Closed()) {
+                // shell is not closed - use simulate to get the end wires
+                TopTools_ListOfShape sim;
+                mkPS.Simulate(2, sim);
+
+                frontwires.push_back(TopoDS::Wire(sim.First()));
+                backwires.push_back(TopoDS::Wire(sim.Last()));
+            }
+        }
+
+        BRepBuilderAPI_MakeSolid mkSolid;
+
+        if (!frontwires.empty()) {
+            // build the end faces, sew the shell and build the final solid
+            TopoDS_Shape front = Part::FaceMakerCheese::makeFace(frontwires);
+            TopoDS_Shape back  = Part::FaceMakerCheese::makeFace(backwires);
+
+            BRepBuilderAPI_Sewing sewer;
+            sewer.SetTolerance(Precision::Confusion());
+            sewer.Add(front);
+            sewer.Add(back);
+
+            for(TopoDS_Shape& s : shells)
+                sewer.Add(s);
+
+            sewer.Perform();
+            mkSolid.Add(TopoDS::Shell(sewer.SewedShape()));
+        } else {
+            // shells are already closed - add them directly
+            for (TopoDS_Shape& s : shells) {
+                mkSolid.Add(TopoDS::Shell(s));
+            }
+        }
+
+        if(!mkSolid.IsDone())
+            return new App::DocumentObjectExecReturn("Result is not a solid");
+
+        TopoDS_Shape result = mkSolid.Shape();
+        BRepClass3d_SolidClassifier SC(result);
+        SC.PerformInfinitePoint(Precision::Confusion());
+        if (SC.State() == TopAbs_IN) {
+            result.Reverse();
+        }
+
+        AddSubShape.setValue(result);
+
+        if(base.IsNull()) {
```

Comment:
Imagive if the below example was a subtractive helix. Then it would chop the cylinder into multiple pieces.
![image](https://user-images.githubusercontent.com/1278189/103453694-c2398e00-4cdc-11eb-94f8-a96ec2573679.png)

Reply:
Excellent question. It is true when you are starting with a sketch on a plane. I don't know if it really should be true in this case, because in that case the sketch should be the base, I think. Something is probably not quite working right. I have looked into this in the past, but I should look again.

Reply:
Yep, this behavior actually causes a bug. When doing subractive helix with no base object (only a sketch on some plane), it will effectively create an additive helix. I will try to find the solution.

Reply:
Before scouting the interwebs for a solution, keep in mind that a subtractive pipe has the same behaviour

Reply:
@vosk. Yes I also figured that out! It is an easy fix. Will make a PR for Pipe later.